swift/example_code/s3/ListBuckets-Simple/Sources/main.swift

```diff
@@ -0,0 +1,60 @@
+/// A simple example that shows how to use the AWS SDK for Swift with the
+/// Amazon Simple Storage Service (Amazon S3) function `ListBuckets`.
+///
+/// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+/// SPDX-License-Identifier: Apache-2.0
+
+// snippet-start:[s3.swift.intro.imports]
+import Foundation
+import ClientRuntime
+import AWSS3
+// snippet-end:[s3.swift.intro.imports]
+
+// snippet-start:[s3.swift.intro.getbucketnames]
+// Return an array containing the names of every available bucket.
+//
+// - Returns: An array of strings listing the available buckets.
+func getBucketNames() async throws -> [String] {
+    // Get an S3Client with which to access Amazon S3.
+    // snippet-start:[s3.swift.intro.client-init]
+    let client = try S3Client(region: "us-east-1")
+    // snippet-end:[s3.swift.intro.client-init]
+
+    // snippet-start:[s3.swift.intro.listbuckets]
+    let output = try await client.listBuckets(
+        input: ListBucketsInput()
+    )
+    // snippet-end:[s3.swift.intro.listbuckets]
+    
+    // Get the bucket names.
+    var bucketNames: [String] = []
+
+    guard let buckets = output.buckets else {
+        return bucketNames
+    }
+    for bucket in buckets {
+        bucketNames.append(bucket.name ?? "<unknown>")
+    }
+
+    return bucketNames
+}
+// snippet-end:[s3.swift.intro.getbucketnames]
+
+// snippet-start:[s3.swift.intro.main]
+/// The program's asynchronous entry point.
+@main
+struct Main {
+    static func main() async {
+        do {
+            let names = try await getBucketNames()
+
+            print("Found \(names.count) buckets:")
+            for name in names {
+                print("  \(name)")
+            }
+        } catch {
```

Comment:
Can you catch AWS- or service-specific error handling?

Reply:
Sort of. This is a work in progress in the SDK right now. I've added some code to do some basic catching of service errors, but it's not very detailed because of stuff not yet implemented in the SDK.